File: scripts/test_solutions/763_partition_labels.py
def partitionLabels(s):
    last = {}
    for i, c in enumerate(s):
        last[c] = i
    result = []
    start = 0
    end = 0
    for i, c in enumerate(s):
        end = max(end, last[c])
        if i == end:
            result.append(end - start + 1)
            start = i + 1
    return result

File: scripts/test_solutions/763_partition_labels.js
var partitionLabels = function(s) {
    const last = {};
    for (let i = 0; i < s.length; i++) last[s[i]] = i;
    const result = [];
    let start = 0, end = 0;
    for (let i = 0; i < s.length; i++) {
        end = Math.max(end, last[s[i]]);
        if (i === end) { result.push(end - start + 1); start = i + 1; }
    }
    return result;
};
